### PR TITLE
Enable analyzers

### PR DESCRIPTION
**Pull request by pakrym**
This is exploratory PR.

I understand that this is a bit too many pragmas.

One of the ideas I had was to add something to mark "Known exposable pubternal type" and stop analyzer from firing on it.

Thoughts? Ideas?

### CODE REVIEW

**Comment by benaadams** [comment]
Can you ignore types with `.Internal` in their namespace for pubternal? (I assume they all do?)

**Comment by pakrym** [comment]
> Can you ignore types with .Internal in their namespace for pubternal? (I assume they all do?)

This analyzer specifically tries to find types with Internal in their namespace exposed through public types.

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs [hunk 1]
```diff
@@ -85,14 +85,18 @@ public virtual TextWriter Output
         /// <summary>
         /// Gets or sets a <see cref="System.Diagnostics.DiagnosticSource"/> instance used to instrument the page execution.
         /// </summary>
+#pragma warning disable PUB0001 // Pubternal type in public API
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs]: I think checking attributes is too much. I'll disable this check.

Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs]: Yeah, attributes aren't a contract. We don't require user code to interact with them.

**Comment by pranavkm** [comment]
Agree with @rynowak - this is super nice. This should be good to go once you fix up the attribute. If you'd like to get this in and follow up cleaning up the suppression around attributes, that should be fine.

**Comment by pakrym** [comment]
Removed checks from attributes.

**Pull request merged**